### ISSUE TITLE
AY2021 SEM 1 Qn4/5 Clarifications

### ISSUE THREAD

**Issue by saiyara1512** [nus-cs2030/2021-s2#391]
Hey guys! Just wanted to double check whether questions 4 (functor) & 5 (LazyList)
<img width="585" alt="Screenshot 2021-04-27 at 18 23 14" src="https://user-images.githubusercontent.com/77091501/116226904-1f72b700-a786-11eb-9b74-319073364d14.png">
from AY2021 sem 1 paper are not in our syllabus for this sem? Especially for LazyList, I am pretty confused on what exactly is tested? Do we just have to know how to read and interpret them (no implementation knowledge needed)?

<img width="567" alt="Screenshot 2021-04-27 at 18 23 03" src="https://user-images.githubusercontent.com/77091501/116226854-1255c800-a786-11eb-91b8-9151aa762fdc.png">

Thanks guys and good luck for tomorrow!

## Screenshots (if any): 
Insert Images here if necessary

**Comment by tzerenchua** [comment]
Hi! You can refer to issues #319 and #335 . Functors is in Lecture 12 while LazyList is in the second half of Lecture 10 and both aren't tested for our sem. 

As for the second half of your question... I guess you could do that for your knowledge/understanding but since they won't be tested for finals I don't think it'll be fair to require us to interpret it

**Comment by brendancjz** [comment]
Yes, Functors and Monads are not tested for Finals! Hmm I believe we need to know how to implement them, aka those map, flatMap, predicate methods that are used in Optional, CompletableFuture and Streams.

**Comment by elizabethh07** [comment]
yes those questions are not tested as they are in Lect 10 (2nd half) and Lect 12.

**Comment by liongthryan** [comment]
I also got a little confused, glad to hear these aren't tested!
